src/util.c

```diff
@@ -398,4 +399,76 @@ void HandleAsking(RaftRedisCommandArray *cmds)
         }
         cmds->commands[0]->argc--;
     }
+}
+
+/* -----------------------------------------------------------------------------
+ * Hashing code - copied directly from Redis.
+ * -------------------------------------------------------------------------- */
+
+/* We have 16384 hash slots. The hash slot of a given key is obtained
+ * as the least significant 14 bits of the crc16 of the key.
+ *
+ * However if the key contains the {...} pattern, only the part between
+ * { and } is hashed. This may be useful in the future to force certain
+ * keys to be in the same node (assuming no resharding is in progress). */
+unsigned int keyHashSlot(const char *key, int keylen) {
+    int s, e; /* start-end indexes of { and } */
+
+    for (s = 0; s < keylen; s++)
+        if (key[s] == '{') break;
+
+    /* No '{' ? Hash the whole key. This is the base case. */
+    if (s == keylen) return crc16_ccitt(key,keylen) & 0x3FFF;
+
+    /* '{' found? Check if we have the corresponding '}'. */
+    for (e = s+1; e < keylen; e++)
+        if (key[e] == '}') break;
+
+    /* No '}' or nothing between {} ? Hash the whole key. */
+    if (e == keylen || e == s+1) return crc16_ccitt(key,keylen) & 0x3FFF;
+
+    /* If we are here there is both a { and a } on its right. Hash
+     * what is in the middle between { and }. */
+    return crc16_ccitt(key+s+1,e-s-1) & 0x3FFF;
+}
+
+RRStatus parseHashSlots(char * slots, char * string)
+{
+    string = strdup(string);
+    RRStatus ret = RR_OK;
+    char *tok = strtok(string, ",");
+    while (tok != NULL) {
+        char * dash = strchr(tok, '-');
+        if (dash == NULL) {
+            char *endptr;
+            unsigned int slot = strtoul(tok, &endptr, 10);
+            if (*endptr != 0 || slot > REDIS_RAFT_HASH_MAX_SLOT) {
+                ret = RR_ERROR;
+                goto exit;
+            }
+            slots[slot] = 1;
+        } else {
+            *dash = 0;
```

Comment:
Minor one, just to make reading easier, it's nice to use same literal type. As this is a string, we can use `'\0'` instead of `0`. Same in other usages, compare with `'\0'` rather than `0` etc.